Build fails

## Description

Latest build jobs fail with:

[e.g.](https://github.com/prateekmedia/appimagepool/actions/runs/5905554857/job/16019868651)
```
Note: intl is pinned to version 0.18.1 by flutter_localizations from the flutter SDK.
See https://dart.dev/go/sdk-version-pinning for details.


Because appimagepool depends on flutter_localizations from sdk which depends on intl
  0.18.1, intl 0.18.1 is required.
So, because appimagepool depends on intl ^0.17.0, version solving failed.

You can try the following suggestion to make the pubspec resolve:
* Try upgrading your constraint on intl: flutter pub add intl:^0.18.1
Error: Process completed with exit code 1.
```

Trying to fix intl with `flutter pub add intl:^0.18.1` then reveals another error:
```
"intl" is already in "dependencies". Will try to update the constraint.
Because appimagepool depends on libadwaita from git which doesn't exist (Could not find
  git ref 'native-controls' (fatal: ambiguous argument 'native-controls': unknown revision
  or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]')), version solving failed.
```

The `native-controls` branch in libadwaita has been deleted after merging [#77](https://github.com/gtk-flutter/libadwaita/pull/77) into libadwaita 2.0.0.

Updating `pubspec.yaml` to use `libadwaita: ^2.0.1` and `intl: ^0.18.1` succeeds but the build now fails because `libadwaita` and its `adwaita` dependency expect Flutter v3:
```
ERROR: ../.pub-cache/hosted/pub.dev/adwaita-0.5.2/lib/src/theme.dart:14:5: Error: No named parameter with the name 'primaryColorDark'.
ERROR:     primaryColorDark: AdwaitaColors.darkBackgroundColor
```

## Proposal

Update Flutter to v3 and dependencies.

I'm not experienced with Flutter but I took a shot at it in #109.

--- a/lib/src/features/home/presentation/home/home_page.dart
+++ b/lib/src/features/home/presentation/home/home_page.dart
@@ -245,18 +245,24 @@ class _HomePageState extends ConsumerState<HomePage> {
                               AdwPreferencesGroup.creditsBuilder(
                                 title: AppLocalizations.of(context)!.authors,
                                 itemCount: developers.length,
-                                itemBuilder: (_, index) => AdwActionRow(
-                                  title: developers[index].name,
-                                  onActivated: developers[index].url.launchIt,
-                                ),
+                                itemBuilder: (_, index) {
+                                  var entry = developers[index];
+                                  return AdwActionRow(
+                                    title: entry.name,
+                                    onActivated: entry.url.launchIt,
+                                  );
+                                },
                               ),
                               AdwPreferencesGroup.creditsBuilder(
                                 title: "Translators",
                                 itemCount: translators.length,
-                                itemBuilder: (_, index) => AdwActionRow(
-                                  title: translators[index].name,
-                                  onActivated: translators[index].url.launchIt,
-                                ),
+                                itemBuilder: (_, index) {
+                                  var entry = translators[index];
+                                  return AdwActionRow(
+                                    title: entry.name,
+                                    onActivated: entry.url.launchIt,
+                                  );
+                                },
                               ),
                             ],
                             appIcon: SvgPicture.network(

--- a/lib/src/features/home/presentation/home/home_page.dart
+++ b/lib/src/features/home/presentation/home/home_page.dart
@@ -242,27 +242,21 @@ class _HomePageState extends ConsumerState<HomePage> {
                             ),
                             appName: appName,
                             credits: [
-                              AdwPreferencesGroup.credits(
+                              AdwPreferencesGroup.creditsBuilder(
                                 title: AppLocalizations.of(context)!.authors,
-                                children:
-                                    List.generate(developers.length, (index) {
-                                  var entry = developers[index];
-                                  return AdwActionRow(
-                                    title: entry.name,
-                                    onActivated: entry.url.launchIt,
-                                  );
-                                }),
+                                itemCount: developers.length,
+                                itemBuilder: (_, index) => AdwActionRow(
+                                  title: developers[index].name,
+                                  onActivated: developers[index].url.launchIt,
+                                ),
                               ),
-                              AdwPreferencesGroup.credits(
+                              AdwPreferencesGroup.creditsBuilder(
                                 title: "Translators",
-                                children:
-                                    List.generate(translators.length, (index) {
-                                  var entry = translators[index];
-                                  return AdwActionRow(
-                                    title: entry.name,
-                                    onActivated: entry.url.launchIt,
-                                  );
-                                }),
+                                itemCount: translators.length,
+                                itemBuilder: (_, index) => AdwActionRow(
+                                  title: translators[index].name,
+                                  onActivated: translators[index].url.launchIt,
+                                ),
                               ),
                             ],
                             appIcon: SvgPicture.network(

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -5,7 +5,7 @@ version: 5.1.0
 publish_to: none
 
 environment:
-  sdk: ">=2.12.0 <3.0.0"
+  sdk: ">=3.1.0 <4.0.0"
 
 dependencies:
   flutter:
@@ -19,15 +19,15 @@ dependencies:
     sdk: flutter
   file_picker: ^5.2.5
   flutter_spinkit: ^5.1.0
-  flutter_svg: ^1.0.0
-  gap: ^2.0.0
+  flutter_svg: ^2.0.6
+  gap: ^3.0.1
   hooks_riverpod: ^2.0.0-dev.9
-  intl: ^0.17.0
-  libadwaita: ^1.2.5
+  intl: ^0.18.1
+  libadwaita: ^2.0.0
   libadwaita_searchbar: ^0.5.0+1
   libadwaita_window_manager:
   lucide_icons: ^0.115.0
-  package_info_plus: ^1.0.6
+  package_info_plus: ^4.0.0
   path_provider: ^2.0.5
   shared_preferences: ^2.0.8
   simple_html_css: ^4.0.0
@@ -40,12 +40,6 @@ dev_dependencies:
   flutter_test:
     sdk: flutter
   flutter_lints: ^2.0.1
-
-dependency_overrides:
-  libadwaita:
-    git:
-      url: https://github.com/gtk-flutter/libadwaita.git
-      ref: native-controls
 
 flutter:
   uses-material-design: true

